packages/react-jss/src/create-use-styles.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+
+import React from 'react'
+import {ThemeContext as DefaultThemeContext} from 'theming'
+
+import {JssContext} from './JssContext'
+import {
+  createStaticSheet,
+  addDynamicRules,
+  removeDynamicRules,
+  updateDynamicRules
+} from './utils/sheets'
+import {getIndex} from './utils/index-counter'
+import type {HookOptions, Styles} from './types'
+import {unmanageSheet, manageSheet} from './utils/managers'
+import {getSheetClasses} from './utils/get-sheet-classes'
+
+const noTheme = {}
+
+function createUseStyles<Theme: {}>(styles: Styles<Theme>, options?: HookOptions<Theme> = {}) {
```

Comment:
`makeStyles` would be shorter?

Reply:
I like this name more TBH. It makes it obvious that a hook is returned. ´makeStyles(styles)` doesn't make much sense. Why should a function make X if I already have X and pass it into `make`?

Reply:
@eps1lon I thought the same.

I also thought about the following syntax:

```js
const {useStyles, withStyles} = makeStyles(styles);
```

You would get hoc with that as well, but this would make it a bigger change once `withStyles` is potentially completely removed.

Reply:
yeah, makeStyles semantically isn't very accurate

Reply:
things which make sense semantically create lower mental overhead, because you don't have to remember what they do

Reply:
> You would get hoc with that as well, 

I would try to clearly separate the HOC interface, because I think HOC interface is going away completely at some point.